### PR TITLE
Fix/facets order

### LINKED ISSUE #2207: BUG? Facets order after selection
**Describe the bug**

A facet widget will always alphabetically order children terms of a selected term, even if order by count was selected.

**Steps to Reproduce**

1. Create a facet widget and configure it to order by Count / Descending
2. Visit the search page
3. See the children terms ordered by count
2. Select the parent term
3. See the children terms now ordered by term name

**Expected behavior**

It is not clear if that is by design or a bug.

### PR DESCRIPTION
**Pull request by oscarssanchez**
### Description of the Change

This PR fixes #2207 

Currently, the problem seems to be https://github.com/10up/ElasticPress/blob/develop/includes/classes/Feature/Facets/Widget.php#L332 sorts alphabetically but doesn't have a logic to discern when ordering by count has been set. I'm adding two new optional parameters to the `order_by_selected` method so it accounts for this scenario.

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: includes/classes/Feature/Facets/Widget.php [hunk 1]
```diff
@@ -332,7 +334,7 @@ private function order_by_selected( $terms, $selected_terms ) {
 		ksort( $selected_terms );
```
[the remark this answers is not in the document]
Comment (oscarssanchez) [reply includes/classes/Feature/Facets/Widget.php]: @Rahmon in this case i think it is worth keeping this when we are sorting by term name, but I forgot to account for ascending. I have added this here: https://github.com/10up/ElasticPress/pull/2288/commits/1b2c0a0928b0ac49d5533b579e4b56c0bc9b17c1 let me know your thoughts

[the remark this answers is not in the document]
Comment (Rahmon) [reply includes/classes/Feature/Facets/Widget.php]: @oscarssanchez When I'm trying to order by term name, the children terms are inverted.

For example setting this way Order Terms By: Term Name and Term Order: Descending, I have this:
1. Cursus
2. Consectetur
   1. xonsectetur children
   2. consectetur children

After selecting Consectetur (parent), the children appears that way:
1. Consectetur
   1. consectetur children
   2. xonsectetur children

The same occurs when I select Term Order: Ascending.

However, it's working for Order Terms By: Count.

[the remark this answers is not in the document]
Comment (oscarssanchez) [reply includes/classes/Feature/Facets/Widget.php]: Thanks @Rahmon i have updated this per our meeting to do the correct ordering.

**Review by Rahmon — DISMISSED** [review]
LGTM :+1:

**Comment by felipeelia** [comment]
@Rahmon @oscarssanchez before we merge this one, a couple of questions:
1. Are we fixing the problem for a +2 tree level? Like
```
Selected Term
   Child Term 1
     Grandchild Term 1
     Grandchild Term 2
   Child Term 2
Other term

```
2. It seems now we have two different functions ordering trees of terms: `Utils\get_term_tree()` and `Facets/Widget::order_by_selected()` . Did you explore the possibility of having only one of them?

**Review by Rahmon — APPROVED** [review]

**Pull request merged**